Gems/Atom/Feature/Common/Assets/Shaders/LightCulling/LightCulling.azsl

```diff
@@ -527,6 +500,48 @@ uint WriteCullingDataToMainMemory(uint lightCount, uint groupIndex, uint3 groupI
     return lightCount;
 }
 
+// Converts depth in view space to depth buffer depth 
+// It is used when depth is in view space and we want to use it as depth for projection matrix
+float ViewSpaceToDepthBuffer(float depth)
+{
+    float a = ViewSrg::m_projectionMatrix[2][3];
+    float b = ViewSrg::m_projectionMatrix[2][2];
+
+    return a / depth - b;
+}
+
+float3 GetViewSpacePosition(float2 tileUv, float depth)
+{
+    // Map UV from [0, 1] to NDC [-1, 1]
+    tileUv = tileUv * 2.0f - 1.0f;
+    float4 clipPos = float4(tileUv.x, -tileUv.y, depth, 1.0f);
+    float4 viewPos = mul(ViewSrg::m_projectionMatrixInverse, clipPos);
+
+    return viewPos.xyz / viewPos.w;
+}
+
+// This function builds an AABB in view space that encompasses the tile
+// The AABB is built by taking the four corners of the tile and the center of the tile, and transforming them to view space
+void BuildAabb(TileLightData tileLightData, uint2 tileId, out float3 aabbCenter, out float3 aabbExtents, out float2 tileCenterUv)
+{
+    float2 tile_uv = float2(tileId) * PassSrg::m_constantData.m_gridPixel;
+    tileCenterUv = tile_uv + PassSrg::m_constantData.m_gridHalfPixel;
+
+    const float depthMin = ViewSpaceToDepthBuffer(-tileLightData.zNear);;
+    const float depthMax = ViewSpaceToDepthBuffer(-tileLightData.zFar);
+
+    float3 pt0 = GetViewSpacePosition(tile_uv, depthMin);
+    float3 pt1 = GetViewSpacePosition(tile_uv, depthMax);
+    float3 pt2 = GetViewSpacePosition(tileCenterUv, depthMin);
+    float3 pt3 = GetViewSpacePosition(tileCenterUv, depthMax);
```

Comment:
should not we use `tile_uv + PassSrg::m_constantData.m_gridPixel` here instead of `tileCenterUv`?

Reply:
You are right, updated the PR after.
Now code matches previous version better.
```
float4 ComputeScreenRays(uint2 tileId, out float2 tileCenterUv)
{
    float2 tile_uv = float2(tileId) * PassSrg::m_constantData.m_gridPixel;

    tileCenterUv = tile_uv + PassSrg::m_constantData.m_gridHalfPixel;

    float4 tileRect;
    tileRect.xy = ScreenUvToRay(tile_uv);
    tileRect.zw = ScreenUvToRay(tile_uv + PassSrg::m_constantData.m_gridPixel);

    return tileRect;
}
```